### PR TITLE
Repair of auth tests

### PR DESCRIPTION
**Pull request by JaurbanRH**
Add sleep into auth0 setup due to slow auth0 API. Add missing cleanup to auth tests. Change scope of browser fixture for devel auth tests.

### CODE REVIEW

File: testsuite/tests/ui/conftest.py [hunk 1]
```diff
@@ -388,10 +388,9 @@ def set_callback_urls(auth0_client):
     Set callback urls for Auth0 application
     """
 
-    @backoff.on_predicate(backoff.fibo, lambda x: not x, 8, jitter=None)
     def _set_callback_urls(client_id, urls: list):
         auth0_client.clients.update(client_id, body={"callbacks": urls})
-        return auth0_client.clients.get(client_id)
+        sleep(15)
```
Comment (pehala) [inline testsuite/tests/ui/conftest.py:393]: This deserves an explanation

**Review by pehala — DISMISSED** [review]

File: testsuite/tests/ui/devel/auth/conftest.py [hunk 2]
```diff
@@ -3,16 +3,48 @@
...
+    """
+    webdriver = SeleniumDriver(source=settings["fixtures"]["ui"]["browser"]["source"],
+                               driver=settings["fixtures"]["ui"]["browser"]["webdriver"],
+                               ssl_verify=settings["ssl_verify"],
+                               remote_url=settings["fixtures"]["ui"]["browser"]["remote_url"],
+                               binary_path=settings["fixtures"]["ui"]["browser"]["binary_path"])
+    webdriver.get_driver()
+    webdriver.post_init()
+    started_browser = ThreeScaleBrowser(selenium=webdriver.webdriver)
+    request.addfinalizer(webdriver.finalize)
```
Comment (mganisin) [inline testsuite/tests/ui/devel/auth/conftest.py:28]: in this case addfinalizer might make sense, however likely 3 lines above?

File: testsuite/tests/ui/auth/conftest.py [hunk 3]
```diff
@@ -74,8 +79,7 @@ def auth0_bounce_login(auth0_setup, navigator, browser, auth0_user):
 
 # pylint: disable=too-many-locals
 @pytest.fixture(scope="module")
-def rhsso_setup(request, custom_admin_login, rhsso_service_info, ui_sso_integration, navigator, testconfig,
-                threescale):
+def rhsso_setup(custom_admin_login, rhsso_service_info, ui_sso_integration, navigator, testconfig, threescale):
```
Comment (mganisin) [inline testsuite/tests/ui/auth/conftest.py:82]: this is almost unrelated change. The goal isn'ŧ to fix "everything", it is about not introducing more "antipaterns". In this particular case, rather than rewrite of finalization it's better to just add missing condition (if I understand the change properly). Keeping commit (and PR) as small as possible containing just relevant changes has highest priority

**Review by mganisin — CHANGES_REQUESTED** [review]

File: testsuite/tests/ui/auth/conftest.py [hunk 4]
```diff
@@ -54,7 +54,12 @@ def auth0_setup(custom_admin_login, testconfig, ui_sso_integration, navigator, s
     assert sso.test_flow_checkbox.is_checked
     sso.publish()
 
-    return urls
+    yield urls
+
+    if not testconfig["skip_cleanup"]:
```
Comment (dhlavac) [inline testsuite/tests/ui/auth/conftest.py:59]: Currently, test are not cleaning Auth0 auth source only users created by this auth source

**Review by dhlavac — DISMISSED** [review]

**Comment by dhlavac** [comment]
IMHO overall would be good to separate Auth0 and RHSSO tests to separate files/modules etc because currently to test only one of them we need to edit tests manually

**Comment by pehala** [comment]
> IMHO overall would be good to separate Auth0 and RHSSO tests to separate files/modules etc because currently to test only one of them we need to edit tests manually

Separating it would probably lead to useless code duplication.

I believe that if the tests are written correctly (Proper cleaning etc.) it shouldn't matter. If it is needed on regular basis then it should be handled with skipif and if it is needed only once in a while I don't think it is that hard to manually.

File: testsuite/tests/ui/conftest.py [hunk 5]
```diff
@@ -388,10 +389,11 @@ def set_callback_urls(auth0_client):
     Set callback urls for Auth0 application
     """
 
-    @backoff.on_predicate(backoff.fibo, lambda x: not x, 8, jitter=None)
     def _set_callback_urls(client_id, urls: list):
         auth0_client.clients.update(client_id, body={"callbacks": urls})
-        return auth0_client.clients.get(client_id)
+        # Due to slow Auth0 API/UI we need to use sleep here. Backoff is not viable option since the client
+        # updates URLs correctly but UI need to wait some time to correctly retrieve URLs.
```
[the remark this answers is not in the document]
Comment (JaurbanRH) [reply testsuite/tests/ui/conftest.py:395]: After trying to login via Auth0, UI shows error message but when you refresh page after couple of seconds everything works as it should. I think that is easier to use sleep instead of using backoff based on is_displayed on UI element and then refreshing page.

**Review by pehala — CHANGES_REQUESTED** [review]

**Review by mganisin — APPROVED** [review]

File: testsuite/tests/ui/devel/auth/conftest.py [hunk 6]
```diff
@@ -3,44 +3,92 @@
...
 from testsuite import settings
+from testsuite.tests.ui import Sessions
+from testsuite.ui.browser import ThreeScaleBrowser
 from testsuite.ui.views.devel.login import LoginDevelView
+from testsuite.ui.webdriver import SeleniumDriver
 
 
 @pytest.fixture(scope="module")
-def custom_devel_auth0_login(browser, navigator, provider_account):
+def browser(request):
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/auth/conftest.py:13]: Isn't this copy-paste?

File: testsuite/tests/ui/conftest.py [hunk 7]
```diff
@@ -347,6 +348,8 @@ def custom_rhsso_login(browser, navigator, threescale, request, testconfig):
     def _login(username, password, rhsso_username=None):
```
Comment (jsmolar) [inline testsuite/tests/ui/conftest.py:348]: What is rhsso_username?

**Review by pehala — APPROVED** [review]
